### ISSUE TITLE
[SDK] grpc-related bugs in Python SDK

### ISSUE THREAD

**Issue by Electronic-Waste** [kubeflow/katib#2395]
### What happened?

![image](https://github.com/user-attachments/assets/78428e7c-6978-4c29-a943-6f0aab28bc40)

![image](https://github.com/user-attachments/assets/85866dbd-ca2f-45e0-a54b-8bc43872013e)

When I called `report_metrics` in the SDK built by myself, the first error above occurred. 

After I modified `grpc.beta.implementations.insecure_channel` to `grpc.insecure_channel`, the second error above occurred.

### What did you expect to happen?

After PR #2344 was merged to the main branch, `beta_create_DBManager_stub` was removed in `api_pb2.py`. But we did not make changes in the following part of codes so that the first error occurred
https://github.com/kubeflow/katib/blob/a6c37e4f3a64ca7f626a4d80f1c3f88362bf03a4/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L1288-L1294

https://github.com/kubeflow/katib/blob/a6c37e4f3a64ca7f626a4d80f1c3f88362bf03a4/sdk/python/v1beta1/kubeflow/katib/api/report_metrics.py#L66-L69

Also grpcio package after `1.0.0` has removed beta version. Sometimes we may get errors when we use beta version.

https://github.com/kubeflow/katib/blob/a6c37e4f3a64ca7f626a4d80f1c3f88362bf03a4/sdk/python/v1beta1/kubeflow/katib/api/report_metrics.py#L57-L59

https://github.com/kubeflow/katib/blob/a6c37e4f3a64ca7f626a4d80f1c3f88362bf03a4/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L1284-L1286

I think we need to may some changes to Python SDK. WDYT👀 @andreyvelich @tenzen-y @johnugeorge 

### Environment

Kubernetes version:
```
Client Version: v1.30.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.1
```
Katib controller version:
```
Built by myself
```
Katib Python SDK version:
```
(Also built by myself)
Name: kubeflow-katib
Version: 0.17.0
Summary: Katib Python SDK for APIVersion v1beta1
Home-page: https://github.com/kubeflow/katib/tree/master/sdk/python/v1beta1
Author: Kubeflow Authors
Author-email: premnath.vel@gmail.com
License: Apache License Version 2.0
Location: /home/xxx/.local/lib/python3.10/site-packages/kubeflow_katib-0.17.0-py3.10.egg
Requires: certifi, grpcio, kubernetes, protobuf, setuptools, six, urllib3
Required-by: 
```

### Impacted by this bug?

Give it a 👍 We prioritize the issues with most 👍

**Comment by Electronic-Waste** [comment]
related PR: #2050 

I think we also need to copy `api_pb2_grpc.py` to Python SDK.

**Comment by Electronic-Waste** [comment]
I encountered this bug while making the push-based metrics collection demo. This issue is also related to the project. 

ref issue: #2340

**Comment by tenzen-y** [comment]
Could you implement the reproducable UTs or E2E test for the `'report_metrics`.
IIUC, `report_metrics` is added without any tests.

**Comment by Electronic-Waste** [comment]
Yes, I think we should also add some UTs or E2E tests for functions in `katib_client.py` like #2325 too.

I guess it will be better to describe the test case problem in another issue and open PRs to it.

WDYT👀 @tenzen-y @andreyvelich @johnugeorge @gaocegege

**Comment by tenzen-y** [comment]
> Yes, I think we should also add some UTs or E2E tests for functions in `katib_client.py` like #2325 too.
> 
> I guess it will be better to describe the test case problem in another issue and open PRs to it.
> 
> WDYT👀 @tenzen-y @andreyvelich @johnugeorge @gaocegege

I have strong objections to evolving the `report_metrics` API without any tests since the `report_metrics` API is new one.
The `report_metrics` can not be handled the same as other Python API.

Before we move forward, we should implement the UTs for the `report_metrcs` so that we prove that the new feature doesn't have regressions. We should not add new features without tests.

Whether the feature doesn't have potential regressions are big deal since Katib is not experimental project, Katib is production project.

**Comment by andreyvelich** [comment]
I agree with @tenzen-y, please let's work on the unit test for `report_metrics` API and for `tune` API first to avoid this problems in the first phase.

**Comment by Electronic-Waste** [comment]
@tenzen-y @andreyvelich UTs for `report_metrics` has been added by now.

I'll work on UTs for `tune` API after this issue is addressed.
